contrib/node/src/python/pants/contrib/node/subsystems/resolvers/npm_resolver.py

```diff
@@ -30,17 +30,46 @@ def register_options(cls, register):
     register(
       '--install-optional', type=bool, default=False, fingerprint=True,
       help='If enabled, install optional dependencies.')
+    register(
+      '--production-only', type=bool, default=False, fingerprint=True,
+      help='If enabled, only install production dependencies')
+    register(
+      '--force', type=bool, default=False, fingerprint=True,
+      help='If enabled, re-download dependencies.')
+    register(
+      '--frozen-lockfile', type=bool, default=True, fingerprint=True,
+      help='If enabled, disallow automatic update of lock files.')
+    register(
+      '--force-option-override', type=bool, default=False, fingerprint=True,
+      help='ADVANCED option. If enabled, options will override hard-coded values.')
+
     NodeResolve.register_resolver_for_type(NodeModule, cls)
 
-  def resolve_target(self, node_task, target, results_dir, node_paths):
-    self._copy_sources(target, results_dir)
+  def resolve_target(self, node_task, target, results_dir, node_paths, resolve_locally=False,
```

Comment:
A method doc would be good here.

In particular, a description somewhere of what the `resolve_locally` argument signifies would be helpful, since it is used all over the place here.